lib/middleware.ts

```diff
@@ -113,6 +113,44 @@ var defaultHandlers = [
   () => ({title: "request-finished"}),
 ];
 
+const defaultContextHandlers = [
+  (req) => ({"request-id": req.headers["x-request-id"]}),
+  (req) => ({"session-id": req.sessionID}),
+];
+
+function handlerData(handlers, ...args) {
+  const data = {};
+  handlers.forEach((h) => {
+    try {
+      const handler_data = h(...args);
+      _.extend(data, handler_data);
```

Comment:
`const` in ES6/TypeScript means that the variable cannot be reassigned. For example:
```ts
const x = 1;
x = 2
```
is not allowed. However properties of objects can be mutated because the value (the reference to the object) is not being changed.